### PR TITLE
records: addition of citations REST

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: inspirehep/modules/records/views.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    def get(self, pid, record):
+        page = request.values.get('page', 1, type=int)
+        size = request.values.get('size', 10, type=int)
+
+        if page < 1 or size < 1:
+            abort(400)
+
+        citing_records_results = record.get_citing_records_query.paginate(
+            page, size, False)
+        citing_records_count = citing_records_results.total
```
Comment (ammirate) [inline inspirehep/modules/records/views.py]: So, from the documentation that I found, `total` will return "the total number of items matching the query (paginate)". I am not sure it is what you want here, probably better use `get_citing_records_query.count()`, wdyt?

File: inspirehep/modules/records/views.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        citing_records_uuids = map(
+            lambda results: results[0], citing_records_results.items)
```
Comment (ammirate) [inline inspirehep/modules/records/views.py]: maybe a list comprehension here would be more pythonic, something like 
`[rec.id for rec in citing_records_results]`

File: inspirehep/modules/records/views.py (hunk 1 above)
Comment (harunurhan) [reply inspirehep/modules/records/views.py]: there is an integration tests for this, so `total` seems to be what we want.

File: inspirehep/modules/records/views.py (hunk 2 above)
Comment (harunurhan) [reply inspirehep/modules/records/views.py]: `[result[0] for result in citing_records_results.items]` is what we want though.

File: inspirehep/modules/records/serializers/json_literature.py [hunk 3]
```diff
@@ -97,3 +99,23 @@ def preprocess_search_hit(self, pid, record_hit, links_factory=None,
...
         return _preprocess_result(result)
+
+
+class LiteratureCitationsJSONSerializer(JSONSerializer):
+
+    def preprocess_record(self, pid, record, links_factory=None, **kwargs):
+        """Prepare a record and persistent identifier for serialization."""
+        return record.dumps()
+
+    def serialize(self, pid, data, links_factory=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (harunurhan) [reply inspirehep/modules/records/serializers/json_literature.py]: just to keep consistent signature for `serialize` method.

File: inspirehep/modules/records/views.py (hunk 2 above)
Comment (ammirate) [reply inspirehep/modules/records/views.py]: Yeah, I was trying to run it, but I was missing the alembic migration and didn't know the `citing_records_results.items` api :+1:

**Comment by ammirate** [comment]
ci test please

File: inspirehep/modules/records/serializers/json_literature.py [hunk 4]
```diff
@@ -97,3 +99,23 @@ def preprocess_search_hit(self, pid, record_hit, links_factory=None,
             preprocess_search_hit(pid, record_hit,
                                   links_factory=links_factory, **kwargs)
         return _preprocess_result(result)
+
+
+class LiteratureCitationsJSONSerializer(JSONSerializer):
+
+    def preprocess_record(self, pid, record, links_factory=None, **kwargs):
+        """Prepare a record and persistent identifier for serialization."""
+        return record.dumps()
```
[the remark this answers is not in the document]
Comment (harunurhan) [reply inspirehep/modules/records/serializers/json_literature.py]: well arguments are there just to consistent signature coming from serializer "interface". And This is actually done not to process anything by overriding inherited method which breaks our thing.

https://github.com/inveniosoftware/invenio-records-rest/blob/c4185c4eac40ccd8c9b532b2ddccce9f91aa63a8/invenio_records_rest/serializers/base.py#L140

**Comment by ammirate** [comment]
ci test please

**Comment by david-caro** [comment]
ci merge please

**Pull request merged**